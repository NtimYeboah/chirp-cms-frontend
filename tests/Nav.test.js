/**
 * Created by ntimobedyeboah on 4/14/17.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import Navbar from './../src/components/Nav';

describe('Navbar snapshot', () => {
    test('It should render the navbar and take a snapshot', () => {
        const component = renderer.create(
            <MemoryRouter>
                <Navbar />
            </MemoryRouter>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Navbar rendered output', () => {
    test('Navbar renders CHIRP in h3', () => {
        const navbar = shallow( <Navbar />);

        expect(navbar.find('h3').text()).toEqual('CHIRP');
        expect(navbar.find('h3').text()).not.toEqual('CHIPR');

    });

    test('Navbar renders 3 children anchor tags', () => {
        const navbar = shallow( <Navbar />);

        expect(navbar.find('div.nav-left').children()).toHaveLength(3);
        expect(navbar.find('div.nav-left').children()).not.toHaveLength(4);
        expect(navbar.find('div.nav-left').childAt(0).text()).toEqual('CHIRP');
        expect(navbar.find('div.nav-left').childAt(0).text()).not.toEqual('CHIPR');
    });
});

describe('Navigating with the rendered navbar menu items', () => {
    test('Navigates to home page when home menu item is clicked', () => {
        const navbar = shallow(<Navbar />);

        navbar.find('div.nav-left').childAt(1).simulate('click');
        expect(location.pathname).toEqual('/');
    });
});