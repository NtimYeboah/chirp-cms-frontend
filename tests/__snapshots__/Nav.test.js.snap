exports[`Navbar snapshot It should render the navbar and take a snapshot 1`] = `
<nav
  className="nav has-shadow">
  <div
    className="container">
    <div
      className="nav-left">
      <a
        className="nav-item">
        <h3
          className="title">
          CHIRP
        </h3>
      </a>
      <a
        className="nav-item is-tab is-hidden-mobile is-active"
        href="/"
        onClick={[Function]}>
        Home
      </a>
      <a
        className="nav-item is-tab is-hidden-mobile"
        href="/posts"
        onClick={[Function]}>
        Posts
      </a>
    </div>
    <span
      className="nav-toggle">
      <span />
      <span />
      <span />
    </span>
  </div>
</nav>
`;
